### PR TITLE
Hide mouse fullscreen

### LINKED ISSUE #3008: Feature: Make the cursor disappear in full screen mode
I don't know why twitch is not making this cursor disappear so if you guys could make this feature in Better TTV would be great to see it !!!

### PR DESCRIPTION
**Pull request by mroseman95**
Sets up a timer that will hide the mouse while in fullscreen mode, and resets that timer after every mouse move.

closes #3008

### CODE REVIEW

**Comment by mroseman95** [comment]
The code I wrote will only hide the mouse cursor when the player is fullscreen, but could easily be changed to hide the mouse whenever it is inactive over the player. Say in theater mode.

I think that option is useful, but decided to just stick to fullscreen, unless others agree that it would make sense to always hide an inactive mouse cursor over the player.

File: src/modules/video_player/index.js [hunk 1]
```diff
@@ -94,6 +104,36 @@ class VideoPlayerModule {
             $(VIDEO_PLAYER_SELECTOR).on('click', '.player-overlay.player-fullscreen-overlay', handlePlayerClick);
         }
     }
+
+    hideMouseFullscreen() {
+        const showMouse = () => {
```
Comment (night) [inline src/modules/video_player/index.js]: these functions don't need to be created here. we should move them outside of this class

File: src/modules/video_player/index.js [hunk 2]
```diff
@@ -94,6 +104,36 @@ class VideoPlayerModule {
...
+        showMouse();
+
+        if (settings.get('hideMouseFullscreen') === true) {
+            hideMouseTimer = setInterval(() => {
+                hideMouse();
+            }, 3000);
+
+            $('body').on('mousemove', '.video-player--fullscreen', () => {
+                showMouse();
+                clearTimeout(hideMouseTimer);
```
Comment (night) [inline src/modules/video_player/index.js]: we should not create a timer that runs to hide the mouse, that is just wrong. you can easily use a debounce function that gets called on mouse move that hides the mouse after a timeout.

File: src/modules/video_player/index.js [hunk 3]
```diff
@@ -94,6 +104,36 @@ class VideoPlayerModule {
...
+            $('.video-player--fullscreen')
+                .toggleClass('bttv-hide-cursor-fullscreen', false);
+        };
+
+        const hideMouse = () => {
+            $('.video-player--fullscreen')
+                .toggleClass('bttv-hide-cursor-fullscreen', true);
+        };
+
+        $('body').off('mousemove', '.video-player--fullscreen');
```
Comment (night) [inline src/modules/video_player/index.js]: create a proper function that adds and removes the listener on page load and setting change.. you're creating a listener on body, which does not need to be recreated on every player load..

File: src/modules/video_player/index.js (hunk 2 above)
Comment (mroseman95) [reply src/modules/video_player/index.js]: I did try that, but an issue arises if a user goes fullscreen without moving their mouse. If they use a hotkey for instance. Then they would have to move their mouse initially for the timer to start.

One solution to this I was thinking of, is adding an event in watcher.js that fires when the screen goes fullscreen, then in the video_player module starting the timer off that.

Comment (night) [reply src/modules/video_player/index.js]: i think requiring mouse movement is an acceptable constraint to hiding the mouse. it's not worth adding additional plumbing for an edge case benefitting only a few people.

Comment (mroseman95) [reply src/modules/video_player/index.js]: Alright, then that should simplify the code a bit.

**Comment by night** [comment]
solved in a32407f1f2d9cb6d4bb0ab5b4e58321a8bcc5727

**Pull request closed without merging**